### PR TITLE
chore(deps): update sqlx to 0.8.2, sea-orm to 1.1, sea-orm-migration to 1.1

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- [x] Use `madsim` org's sqlx instead.

Closes https://github.com/risingwavelabs/risingwave/issues/18853.

This PR updates sqlx to v0.8.2 (https://github.com/risingwavelabs/sqlx/pull/2). We have to update it sooner or later, since other projects e.g. iceberg-rs depend on it.

We also have to bump `sea-orm`, because it depends on `sqlx`. Otherwise the `sqlx` version it uses will have conflicts.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/model/migration/src/m20230908_072257_init.rs [hunk 1]
```diff
@@ -378,7 +378,7 @@ impl MigrationTrait for Migration {
                     )
                     .col(
                         ColumnDef::new(Fragment::StreamNode)
-                            .blob(BlobSize::Long)
+                            .blob()
```
Comment (kwannoel) [inline src/meta/model/migration/src/m20230908_072257_init.rs:381]: It's the result of  breaking changes from `sea-query`: https://github.com/SeaQL/sea-query/blob/master/CHANGELOG.md#new-features-1.

cc @yezizp2012 wonder if it's safe to change the migration scripts.

Comment (yezizp2012) [reply src/meta/model/migration/src/m20230908_072257_init.rs:381]: The blob size is required by mysql backend. `MySQL` has four `BLOB` types: `TINYBLOB`, `BLOB`, `MEDIUMBLOB`, and `LONGBLOB`, every type has a corresponding length limit. After the breaking changes of `sea-query`, I think we need to specify the column type as `.custom(MySqlType::LongBlob)` if the backend is `MySQL`. 

FYI: both `PostgreSQL` and `SQLite` have a 1GB size limit for the `BLOB` type.

Comment (kwannoel) [reply src/meta/model/migration/src/m20230908_072257_init.rs]: Updated it https://github.com/risingwavelabs/risingwave/pull/19145/commits/77707e1b4f5924b0ed9c447ada8f38d2e24d3c97

File: Cargo.lock [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
-version = 3
+version = 4
```
Comment (kwannoel) [inline Cargo.lock:3]: Not sure if this is safe to update? This was update automatically.

File: Cargo.toml [hunk 3]
```diff
@@ -351,11 +351,9 @@ getrandom = { git = "https://github.com/madsim-rs/getrandom.git", rev = "e79a7ae
...
 tokio-retry = { git = "https://github.com/madsim-rs/rust-tokio-retry.git", rev = "95e2fd3" }
 tokio-postgres = { git = "https://github.com/madsim-rs/rust-postgres.git", rev = "ac00d88" }
-
-# sqlx version: v0.7.4
-# patch diffs: https://github.com/madsim-rs/sqlx/pull/3
-sqlx = { git = "https://github.com/risingwavelabs/sqlx.git", rev = "ff6d6d2dc0e9e8e47282fd29be006eed7ae3421a" }
-
+# sqlx version: v0.8.2
+# patch diffs: https://github.com/madsim-rs/sqlx/pull/4
+sqlx = { git = "https://github.com/madsim-rs/sqlx.git", rev = "3efe6d0065963db2a2b7f30dee69f647e28dec81" }
```
Comment (kwannoel) [inline Cargo.toml:356]: Changed to `madsim-rs` org. Previously I missed out updating the repository.

**Review by yezizp2012 — APPROVED** [review]

**Review by cyliu0 — APPROVED** [review]

File: src/meta/model/migration/src/m20230908_072257_init.rs (hunk 1 above)
Comment (BugenZhao) [reply src/meta/model/migration/src/m20230908_072257_init.rs]: Done in #19207

**Review by BugenZhao — APPROVED** [review]

**Comment by kwannoel** [comment]
Skipped deterministic fuzz tests.
These test failure are due to the `sqrt negative number` failure (can be ignored), and a separate bug:
```
thread '<unnamed>' panicked at src/common/src/util/sort_util.rs:303:74:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: rust_begin_unwind
             at /rustc/52fd9983996d9fcfb719749838336be66dee68f9/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /rustc/52fd9983996d9fcfb719749838336be66dee68f9/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /rustc/52fd9983996d9fcfb719749838336be66dee68f9/library/core/src/panicking.rs:148:5
   3: core::option::unwrap_failed
             at /rustc/52fd9983996d9fcfb719749838336be66dee68f9/library/core/src/option.rs:2004:5
   4: unwrap<&risingwave_common::catalog::schema::Field>
             at /rustc/52fd9983996d9fcfb719749838336be66dee68f9/library/core/src/option.rs:969:21
   5: fmt
             at ./src/common/src/util/sort_util.rs:303:74
   6: to_string<risingwave_common::util::sort_util::ColumnOrderDisplay>
             at /rustc/52fd9983996d9fcfb719749838336be66dee68f9/library/alloc/src/string.rs:2682:9
   7: display<risingwave_common::util::sort_util::ColumnOrderDisplay>
             at ./.cargo/registry/src/index.crates.io-6f17d22bba15001f/pretty-xmlish-0.1.13/src/lib.rs:136:9
   8: {closure#0}
             at ./src/frontend/src/optimizer/property/order.rs:72:13
```

Which is independent of this PR.

**Pull request merged**